pkg/service/types.go

```diff
@@ -43,9 +50,21 @@ type MeshService struct {
 
 	// Protocol is the protocol served by the service's port
 	Protocol string
+
+	// ProviderKey represents the name of the original entity from which this MeshService was created (e.g. a Kubernetes service)
+	ProviderKey string
+
```

Comment:
In that scenario would a providerkey ever be different than the name?

Reply:
I'm assuming you mean "what scenario" - headless services are the one that comes to mind. I feel like `MeshService.Name` should have semantic meaning beyond just the original service name in the provider. I think it also should identify a _routeable_ entry within the mesh's catalog (which is why I created ProviderKey). Just because the provider stored the service with one name doesn't mean the mesh catalog entry should have that exact same name.

Reply:
I see MeshService is sort of already decoupled, since it's our own custom type, as opposed to something out of K8s, or even Envoy, so I don't think we need to make further abstractions around it.

Reply:
I'm assuming this is meant for the other thread? If so, that makes sense to me, and I'll remove the interface

Reply:
yup, feel free to resolve